app/javascript/components/student/Editor.tsx

```diff
@@ -37,30 +105,63 @@ type State = {
   status?: EditorStatus
 }
 
+enum ActionType {
```

Comment:
I just realized that the `EditorStatus` does not map 1-1 to our dispatch actions. I created a new type solely for our dispatch actions.